tapestry.js

```diff
@@ -975,39 +992,109 @@ function getBoundedCoord(coord, maxCoord) {
     return Math.max(MAX_RADIUS, Math.min(maxCoord - MAX_RADIUS, coord));
 }
 
-function getChildren(id) {
-    var children = [];
-    var dataLinks = dataset.links;
-    for (var linkId in dataLinks) {
-        var link = dataLinks[linkId];
-        //SEARCH FOR: CURRENT_ID_NODE -> CHILD links
-        if (typeof link.source === 'number' && link.source === id) {
-            children.push(link.target);
-        } else if (typeof link.source === 'object' && link.source.id === id) {
-            children.push(link.target.id);
-        }
+/* Add 'depth' parameter to each node recursively. 
+   The depth is determined by the number of levels from the root each node is. */
+
+function addDepthToNodes(rootId, depth, visited) {
+    visited.push(rootId);
+
+    var depthAt = 0;
+
+    dataset.nodes[findNodeIndex(rootId)].depth = depth;
+    var children = getChildren(rootId, 1);
+
+    var childLevel;
+
+    while (depthAt < children.length) {
```

Comment:
Can you add some comments in the code to describe what you are doing at each loop / step? It would also help if you change your variable names to be more descriptive. For example, `depthAt` is a bit ambiguous. You could also add a comment to describe what each variable is supposed to keep where you declare the variable.